### PR TITLE
hibernate relations done

### PR DESCRIPTION
**Pull request by not4und**

### CODE REVIEW

File: src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java [hunk 1]
```diff
@@ -12,11 +15,33 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
...
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(actor);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Couldn't add actor to actorsDB." + actor, e);
```
Comment (Rommelua) [inline src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java:29]:
```suggestion
            throw new DataProcessingException("Couldn't add actor to DB." + actor, e);
```

File: src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java [hunk 2]
```diff
@@ -12,11 +15,33 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
...
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(country);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Couldn't add country to countriesDB." + country, e);
```
Comment (Rommelua) [inline src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java:29]:
```suggestion
            throw new DataProcessingException("Couldn't add country to DB." + country, e);
```

File: src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java [hunk 3]
```diff
@@ -12,11 +15,33 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
...
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Couldn't add movie to moviesDB." + movie, e);
```
Comment (Rommelua) [inline src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java:29]:
```suggestion
            throw new DataProcessingException("Couldn't add movie to DB." + movie, e);
```

File: src/main/java/mate/academy/hibernate/relations/exception/DataProcessingException.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package mate.academy.hibernate.relations.exception;
+
+public class DataProcessingException extends RuntimeException {
+    public DataProcessingException(String message) {
+        super(message);
+    }
```
Comment (Rommelua) [inline src/main/java/mate/academy/hibernate/relations/exception/DataProcessingException.java:6]: Why do we need this one?

**Review by Rommelua — CHANGES_REQUESTED** [review]
Good work, I have a few minor comments.

File: src/main/java/mate/academy/hibernate/relations/service/impl/MovieServiceImpl.java [hunk 5]
```diff
@@ -1,16 +1,23 @@
...
     @Override
     public Movie add(Movie movie) {
-        return null;
+        return movieDao.add(movie);
     }
 
     @Override
     public Movie get(Long id) {
-        return null;
+        return movieDao.get(id).orElseThrow();
```
Comment (kateryna-deshchinska) [inline src/main/java/mate/academy/hibernate/relations/service/impl/MovieServiceImpl.java:21]: add some informative exception

File: src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java [hunk 6]
```diff
@@ -1,16 +1,23 @@
...
     @Override
     public Country add(Country country) {
-        return null;
+        return countryDao.add(country);
     }
 
     @Override
     public Country get(Long id) {
-        return null;
+        return countryDao.get(id).orElseThrow();
```
Comment (kateryna-deshchinska) [inline src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java:21]: Add informative RuntimeException

File: src/main/resources/hibernate.cfg.xml [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+        <property name="connection.username">root</property>
+        <property name="connection.password">root</property>
+        <property name="show_sql">true</property>
+        <property name="hbm2ddl.auto">create-drop</property>
+
+        <mapping class="mate.academy.hibernate.relations.model.Actor" />
+        <mapping class="mate.academy.hibernate.relations.model.Country" />
+        <mapping class="mate.academy.hibernate.relations.model.Movie" />
+    </session-factory>
+</hibernate-configuration>
```
Comment (kateryna-deshchinska) [inline src/main/resources/hibernate.cfg.xml:15]: add empty line

**Review by kateryna-deshchinska — APPROVED** [review]

**Pull request closed without merging**